src/cassandra/future.rs

```diff
@@ -179,15 +182,22 @@ impl<T: Completable> futures::Future for CassFuture<T> {
                 // Install the callback. If callback cannot be sent, report immediate `Err`.
                 let data =
                     (self.state.as_ref() as *const FutureTarget) as *mut ::std::os::raw::c_void;
-                cass_future_set_callback(self.inner, Some(notify_task), data).to_result(())
-            } else {
-                Ok(())
+                cass_future_set_callback(self.inner, Some(notify_task), data).to_result(())?;
             }
-            .and_then(move |_| ret)
+
+            ret
         }
     }
 }
 
+impl<T: Completable> CassFuture<T> {
+    /// Synchronously executes the CassFuture, blocking until it
+    /// completes.
+    pub fn wait(self) -> Result<T> {
+        unsafe { get_completion(self.inner) }
```

Comment:
This changes the implementation so that the waiting happens purely in C, rather than by a driving loop within `futures`. That's fine, but it relies on knowing that `get_completion` supports such use. It does, because `cass_future_get_error_code` does, but that may not be obvious. Please can you fix the comments on lines 201 and 203 that say "a completed future" and "Future is ready" to reflect the new reality?